src/message_interface/message_handler.rs

```diff
@@ -535,7 +539,17 @@ impl WalletMessageHandler {
             }
             AccountMethod::BurnNft { nft_id, options } => {
                 convert_async_panics(|| async {
-                    let transaction = account_handle.burn_nft(NftId::try_from(&nft_id)?, options).await?;
+                    let transaction = account_handle
+                        .burn_nft(
+                            NftId::try_from(&nft_id)?,
+                            match options {
```

Comment:
For this and all of them, looks like you could just map instead of match ?

Reply:
But with the map I can't do the error handling? I think there was some way to do the same with less code, but couldn't find it

Reply:
Yes that was what I searched, thanks!